packages/salesforcedx-vscode-core/src/commands/templates/metadataTypeConstants.ts

```diff
@@ -1,10 +1,12 @@
 export const APEX_CLASS_DIRECTORY = 'classes';
 export const APEX_CLASS_EXTENSION = '.cls';
 export const APEX_CLASS_TYPE = 'apexclass';
+export const APEX_CLASS_NAME_MAX_LENGTH = 40;
```

Comment:
this could be fetched at build time from an API that has the limits? Maybe, but since this value is rarely changed, so it is not worthy automating it.